RecoPixelVertexing/PixelTriplets/plugins/CACell.h

```diff
@@ -0,0 +1,233 @@
+#ifndef CACELL_H_
+#define CACELL_H_
+
+#include "RecoTracker/TkHitPairs/interface/RecHitsSortedInPhi.h"
+#include "TrackingTools/TransientTrackingRecHit/interface/SeedingLayerSetsHits.h"
+#include "RecoTracker/TkTrackingRegions/interface/TrackingRegion.h"
+
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+#include <cmath>
+#include <array>
+
+class CACell {
+public:
+    using Hit = RecHitsSortedInPhi::Hit;
+    using CAntuplet = std::vector<CACell*>;
+
+    CACell(const HitDoublets* doublets, int doubletId, const unsigned int cellId, const int innerHitId, const int outerHitId) :
+    theCAState(0), theInnerHitId(innerHitId), theOuterHitId(outerHitId), theCellId(cellId), hasSameStateNeighbors(0), theDoublets(doublets), theDoubletId(doubletId),
+    theInnerR(doublets->r(doubletId, HitDoublets::inner)), theOuterR(doublets->r(doubletId, HitDoublets::outer)),
+    theInnerZ(doublets->z(doubletId, HitDoublets::inner)), theOuterZ(doublets->z(doubletId, HitDoublets::outer)) {
+    }
+
+    unsigned int get_cell_id() const {
```

Comment:
bouncing caps is a more typical CMSSW method naming style.
underscore-separated in ~~ OK.
In case you get to refactoring.